assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -645,45 +658,44 @@ const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
 
   x += spaceBetween + SLOT_WIDTH;
 
-  const [isDone, setIsDone] = useState(false);
-
   // Get the size of the diagram svg, excluding emphasis
   let dimensions = document.getElementById("line-map")?.getBoundingClientRect();
   let height = dimensions?.height ?? 0;
   let width = dimensions?.width ?? 0;
 
   useEffect(() => {
-    // Get updated dimensions each time this hook runs
-    dimensions = document.getElementById("line-map")?.getBoundingClientRect();
-    height = dimensions?.height ?? 0;
-    width = dimensions?.width ?? 0;
-
-    if (svgHeight != 0 && width && height) {
-      // if scaleFactor has already been applied to the line-map, we need to reverse that
-      height = height / scaleFactor;
-      width = width / scaleFactor;
-
-      // First, scale x. Then, check if it needs abbreviating. Then scale y, given the abbreviation
-      const xScaleFactor = 904 / width;
-
-      const needsAbbreviating =
-        height * xScaleFactor + getEmphasisHeight(xScaleFactor) > svgHeight &&
-        !doAbbreviate;
-      if (needsAbbreviating) {
-        setDoAbbreviate(true);
-        // now scale y, which requires re-running this effect
-      } else {
-        const yScaleFactor = (svgHeight - getEmphasisHeight(1)) / height
-        const factor = Math.min(
-          xScaleFactor,
-          yScaleFactor
-        );
-        setScaleFactor(factor);
-        setTimeout(() => {
-          setIsDone(true);
-        }, 200);
+    setTimeout(() => {
+      // Get updated dimensions each time this hook runs
```

Comment:
The nested callbacks are getting a little hard to follow. Can you make the original callback its own function and call that function in the `setTimeout`?

Reply:
Good call. Did this help?